examples/named-worker-with-on-fail/named_worker_with_on_fail.bal

```diff
@@ -0,0 +1,28 @@
+import ballerina/io;
+
+public function main() {
+    int[] vals = [2, 3, 4, 5];
+    int key = 1;
+    int val = 0;
+
+    worker w1 {
+        int? index = vals.indexOf(key);
+        if index != () {
+            val = vals[index];
+        } else {
+            check error("value not found");
+        }
+        val -> w2;
+    } on fail {
+        // handle the error thrown in the worker body
+        -1 -> w2;
+    }
+
+    worker w2 returns int|error:NoMessage {
+        int|error:NoMessage result = <- w1 | w1;
```

Comment:
Oops. Since we need `alternate-receive` here. The order mentioned here would not work. I will update the new order in the issue itself.

Reply:
New order: https://github.com/ballerina-platform/ballerina-dev-website/issues/8856#issuecomment-2044468508